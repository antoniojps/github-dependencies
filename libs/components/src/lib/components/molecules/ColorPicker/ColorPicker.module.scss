.picker {
  height: 36px;
  width: 36px;
  border: 1px solid var(--border);
  border-radius: var(--radius);
  background-color: var(--background);
  cursor: pointer;
  &:active {
    opacity: 0.9;
  }
}

.small {
  height: 18px;
  width: 18px;
}
