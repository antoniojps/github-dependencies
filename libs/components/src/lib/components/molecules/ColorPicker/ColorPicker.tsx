import React, { useState, useEffect } from 'react';
import styles from './ColorPicker.module.scss';
import { Popover, usePopover } from '../';
import { SketchPicker, RGBColor } from 'react-color';
import { parseToRgb, rgbToColorString } from 'polished';
import { noop } from 'lodash';
import classNames from 'classnames';

type ColorPickerProps = {
  /* Selected color  */
  color: string;
  size?: 'medium' | 'small';
  onChangeColor?: (color: string) => void;
};

function parseToRGBColor(color: string) {
  const parsed = parseToRgb(color);
  const alpha = 'alpha' in parsed ? parsed.alpha : 1;

  return {
    a: alpha,
    b: parsed.blue,
    g: parsed.green,
    r: parsed.red,
  };
}

function parseToColorString(color: RGBColor) {
  const converted = {
    blue: color.b,
    red: color.r,
    green: color.g,
    alpha: 'a' in color ? color.a : 1,
  };

  return rgbToColorString(converted);
}

export const ColorPicker = ({ color, onChangeColor = noop, size = 'medium' }: ColorPickerProps) => {
  const [isOpen, toggle] = usePopover(false);
  const [pickedRgb, setPickedRgb] = useState<RGBColor>(parseToRGBColor(color));

  useEffect(() => {
    setPickedRgb(parseToRGBColor(color));
  }, [color]);

  const handleChangeComplete = (rgb: RGBColor) => {
    const colorString = parseToColorString(rgb);
    onChangeColor(colorString);
  };

  return (
    <div>
      <Popover
        content={
          <SketchPicker
            color={pickedRgb}
            onChange={(color) => setPickedRgb(color.rgb)}
            onChangeComplete={(color) => handleChangeComplete(color.rgb)}
          />
        }
        isOpen={isOpen}
        toggle={toggle}
        positions={['bottom']}
        align={'start'}
        padding={5}
      >
        <div
          className={classNames(styles.picker, { [styles.small]: size === 'small' })}
          style={{ backgroundColor: parseToColorString(pickedRgb) }}
          role="button"
        />
      </Popover>
    </div>
  );
};
