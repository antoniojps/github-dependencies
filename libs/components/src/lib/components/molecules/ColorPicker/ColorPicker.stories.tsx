import React, { ReactElement } from 'react';
import { ColorPicker } from './ColorPicker';
import { Text, Spacer } from '@geist-ui/react';

export default {
  title: 'molecules/ColorPicker',
};

export const Basic = (): ReactElement => {
  const [selectedColor, setSelectedColor] = React.useState('yellow');

  return (
    <div>
      <ColorPicker color={selectedColor} onChangeColor={setSelectedColor} />
      <Spacer y={0.5} />
      <Text h1 style={{ color: selectedColor }}>
        Change my color
      </Text>
    </div>
  );
};

export const Small = (): ReactElement => {
  const [selectedColor, setSelectedColor] = React.useState('yellow');

  return (
    <div>
      <ColorPicker color={selectedColor} onChangeColor={setSelectedColor} size="small" />
      <Spacer y={0.5} />
      <Text h1 style={{ color: selectedColor }}>
        Change my color
      </Text>
    </div>
  );
};
