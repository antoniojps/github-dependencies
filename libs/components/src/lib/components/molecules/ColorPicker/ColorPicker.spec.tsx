import React from 'react';
import { render } from '@testing-library/react';

import { ColorPicker } from './ColorPicker';

describe('ColorPicker', () => {
  it('renders', () => {
    const { asFragment } = render(<ColorPicker color="yellow" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders small', () => {
    const { asFragment } = render(<ColorPicker color="yellow" size="small" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
