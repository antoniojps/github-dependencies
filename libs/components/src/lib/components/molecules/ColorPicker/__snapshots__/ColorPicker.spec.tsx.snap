// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="children"
    >
      <div
        class="picker"
        role="button"
        style="background-color: rgb(255, 255, 0);"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ColorPicker renders small 1`] = `
<DocumentFragment>
  <div>
    <div
      class="children"
    >
      <div
        class="picker small"
        role="button"
        style="background-color: rgb(255, 255, 0);"
      />
    </div>
  </div>
</DocumentFragment>
`;
